### PR TITLE
AddIfChildDirective is optionally case sensitive

### LINKED ISSUE #194: AddIfChildDirective should support case sensitivity
### Expected Behavior
AddIfChildDirective can be case sensitive

### Actual Behavior
AddIfChildDirective always ignores case

### PR DESCRIPTION
**Pull request by fneu**
### Please check if the PR fulfills these requirements
- [x] I made sure that my code builds
- [x] I merged the master into this branch before pushing
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

### What kind of change does this PR introduce?** (check one with "x")
- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

### What is the current behavior? (You can also link to an open issue here)
closes #194 

### What is the new behavior?
added boolean param `ignoreCase` to AddIfChild directive

### Does this PR introduce a breaking change? (check one with "x")
- [ ] Yes
- [x] No

###  Other information

### CODE REVIEW

**Review by HerickJ-22 — CHANGES_REQUESTED** [review]
The new ignoreCase parameter also has to be added to 
https://github.com/icarus-consulting/Yaapii.Xambly/blob/99c5d85a1de2ab42237da56cb1a2617749879be4/src/Yaapii.Xambly/Directive/Directives.cs#L307

File: tests/Test.Yaapii.Xambly/Directive/AddIfChildDirectiveTests.cs [hunk 1]
```diff
@@ -53,5 +53,33 @@ public void DoesntAddTwice()
...
+                        new PushDirective(),
+                        new AddDirective("item"),
+                        new AddDirective("id"),
+                        new SetDirective("some-id"),
+                        new PopDirective(),
+                        new AddIfChildDirective("item", "id", "some-id"),
+                        new AddDirective("child"),
+                        new UpDirective(),
+                        new UpDirective(),
+                        new AddIfChildDirective("item", "id", "SOME-ID", false),
```
Comment (pz1337) [inline tests/Test.Yaapii.Xambly/Directive/AddIfChildDirectiveTests.cs:75]: i think this test would have been green with the previous version (without the false), wouldn't it?
i think what you want to test is:
```
Assert.False(
    ...
    new AddIfChildDirective("item", "id", "some-ID", false),
    ...
```

**Review by pz1337 — DISMISSED** [review]

File: tests/Test.Yaapii.Xambly/Directive/AddIfChildDirectiveTests.cs (hunk 1 above)
Comment (fneu) [reply tests/Test.Yaapii.Xambly/Directive/AddIfChildDirectiveTests.cs:75]: No. The trick here is that we have two `<item>`s with different ids. Without passing `false` here, the second would not have been created and both `<child>`s would have been created in the first `<item>` with `<id>some-id</id>` (lowercase).

Comment (pz1337) [reply tests/Test.Yaapii.Xambly/Directive/AddIfChildDirectiveTests.cs:75]: understood now, thanks 👍

**Review by pz1337 — APPROVED** [review]

**Pull request merged**